client/state/data-layer/wpcom/sites/activity/index.js

```diff
@@ -63,7 +63,7 @@ export const handleActivityLogRequest = ( { dispatch }, action ) => {
 };
 
 export const receiveActivityLog = ( { dispatch }, action, data ) => {
-	dispatch( activityLogUpdate( action.siteId, fromApi( data ), data.totalItems, action.params ) );
+	dispatch( activityLogUpdate( action.siteId, data, data.totalItems, action.params ) );
 };
 
 export const receiveActivityLogError = ( { dispatch }, { siteId }, error ) => {
```

Comment:
We now have access to any validation failures via the `error` here, is that right? We may want to at least `debug` that to the console, if not `console.warn`, because it seems it is a valuable indication that there is a misunderstanding somewhere between the server and the client.

Unless the data-layer already (or plans to) handle these alerts, as mentioned previously in the PR.

Reply:
I don't believe the data-layer does anything specific, it's left up to us.

I agree that we should make this accessible somehow. As-is, this very closely reflects the existing implementation. I'd like to follow-up with improved error handling.

Reply:
The general recommendation I have been passing along is to structure data layer files with a success responder like `addWidget()` which injects data into our state tree and an `announceFailure()` which creates an `errorNotice()` to alert the customer of a failure. In most cases I think simply announcing that it failed is probably enough to surface the problems and fix them.